backend/store/etcd/watcher.go

```diff
@@ -0,0 +1,178 @@
+package etcd
+
+import (
+	"context"
+	"errors"
+	"strings"
+	"sync"
+
+	"github.com/coreos/etcd/clientv3"
+	"github.com/sensu/sensu-go/backend/store"
+)
+
+// watcher implements the store.Watcher interface
+type watcher struct {
```

Comment:
So the idea here was that the channel returned by the [`Watch` method](https://github.com/etcd-io/etcd/blob/e34423b6fff5bd14d7f765944273ed2eee250f3c/clientv3/watch.go#L72) would return a single event (`store.WatchEvent`) instead of a [list of events](https://github.com/etcd-io/etcd/blob/e34423b6fff5bd14d7f765944273ed2eee250f3c/clientv3/watch.go#L83) so the consumer doesn't have to deal with raw events from etcd, but if you feel that's unnecessary I'm totally okay with implementing the etcd interface.

Reply:
Ah sorry, I was asking you to add some comments to the code explaining why it's useful. Something similar to what you posted here. :)

Reply:
Oh yes absolutely, my bad haha!